public/app/features/scenes/dashboard/DashboardsLoader.test.ts

```diff
@@ -1,13 +1,352 @@
+      // @ts-expect-error - private
+      expect(loader.state.dashboard?.urlSyncManager).toBeDefined();
+      expect(loader.state.isLoading).toBe(false);
+    });
+  });
+
+  describe('when creating dashboard scene', () => {
+    it('should initialize the DashboardScene with the model state', () => {
+      const dash = {
+        ...defaultDashboard,
+        title: 'test',
+        uid: 'test-uid',
+        time: { from: 'now-10h', to: 'now' },
+        templating: {
+          list: [
+            {
+              hide: 2,
+              name: 'constant',
+              skipUrlSync: false,
+              type: 'constant' as VariableType,
+              rootStateKey: 'N4XLmH5Vz',
+              query: 'test',
+              id: 'constant',
+              global: false,
+              index: 3,
+              state: LoadingState.Done,
+              error: null,
+              description: '',
+              datasource: null,
+            },
+          ],
+        },
+      };
+      const oldModel = new DashboardModel(dash);
+
+      const scene = createDashboardSceneFromDashboardModel(oldModel);
+
+      expect(scene.state.title).toBe('test');
+      expect(scene.state.uid).toBe('test-uid');
+      expect(scene.state?.$timeRange?.state.value.raw).toEqual(dash.time);
+      expect(scene.state?.$variables?.state.variables).toHaveLength(1);
+      expect(scene.state.subMenu).toBeDefined();
+    });
+  });
+
+  describe('when organizing panels as scene children', () => {
+    it('should create panels within collapsed rows', () => {
+      const panel: Panel = {
+        title: 'test',
+        type: 'timeseries',
+        gridPos: { x: 1, y: 0, w: 12, h: 8 },
+        fieldConfig: {
+          defaults: {},
+          overrides: [],
+        },
+        options: {
+          defaults: {},
+          overrides: [],
+        },
+        transformations: [],
+        transparent: false,
+        repeatDirection: 'h',
+      };
+
+      const row: RowPanel = {
+        title: 'test',
+        type: 'row',
+        gridPos: { x: 0, y: 0, w: 12, h: 1 },
+        collapsed: true,
+        id: 1,
+        panels: [panel],
+      };
+
+      const dashboard = {
+        ...defaultDashboard,
+        panels: [row],
+      };
+
+      const oldModel = new DashboardModel(dashboard);
+
+      const scene = createDashboardSceneFromDashboardModel(oldModel);
+
+      expect(scene.state.body.state.children).toHaveLength(1);
+      const rowScene = scene.state.body.state.children[0] as SceneGridRow;
+      expect(rowScene).toBeInstanceOf(SceneGridRow);
+      expect(rowScene.state.title).toEqual(row.title);
+      expect(rowScene.state.placement?.y).toEqual(row.gridPos!.y);
+      expect(rowScene.state.isCollapsed).toEqual(row.collapsed);
+      expect(rowScene.state.children).toHaveLength(1);
+      expect(rowScene.state.children[0]).toBeInstanceOf(VizPanel);
+    });
+
+    it('should create panels within expanded row', () => {
```

Comment:
The setup (state) for the panels is pretty long with lots of things that are not relevant for the test. think a function that takes a partial Panel would help here, so we can only specify the gridPos

Reply:
Agree, we have a helper for tat that should be useful: [createPanelJSONFixture](https://github.com/grafana/grafana/blob/main/public/app/features/dashboard/state/__fixtures__/dashboardFixtures.ts#L32)